src/grpc/user.rs

```diff
@@ -120,7 +121,18 @@ impl user_server::User for UserService {
 
         match Vote::get_all_by_client_hash(&client_hash, Some(snap_id), conn).await {
             Ok(votes) => {
-                let votes = votes.into_iter().map(|vote| vote.into()).collect();
+                let votes = try_join_all(votes.into_iter().map(|vote| async {
+                    let snap_name = get_snap_name(
+                        &vote.snap_id,
+                        &self.ctx.config.snapcraft_io_uri,
+                        &self.ctx.http_client,
+                    )
+                    .await?;
+
+                    Result::<PbVote, Error>::Ok(PbVote::from_vote_and_snap_name(vote, &snap_name))
+                }))
+                .await
+                .map_err(|_| Status::unknown("Internal server error"))?;
```

Comment:
You can probably avoid the explicit type annotations for the Result here if you instead just use `try_join_all` to collect the snap names and then map over that to build out the votes.

Reply:
Hm, I don't have access to the `vote` in a subsequent `map` over the names, though

Reply:
ah sorry, I missed that. You'd need to zip the votes with the names and at that point its not much of a readability win anymore I don't think. Feel free to ignore the suggestion :slightly_smiling_face: